### ISSUE TITLE
sending interrupt signal while removing a package leaves lock file

### ISSUE THREAD

**Issue by despairblue** [aurapm/aura#319]
## Steps to reproduce
1. install somepackage
2. remove it `sudo aura -R ${somepackage}`
3. when asked if you're really sure send interrupt by using `ctlr+c`
4. if you try to remove it again you get:

```
error: failed to init transaction (unable to lock database)
error: could not lock database: File exists
  if you're sure a package manager is not already
  running, you can remove /var/lib/pacman/db.lck
aura >>= Please check your input.
```

It does not always happen but approximately one third of the time.

Also it might have something to do with powerpill since I got this error (but only once):

```
checking dependencies...

Packages (1) me-tv-1:1.4.0.12-3

Total Removed Size:  1.96 MiB

:: Do you want to remove these packages? [Y/n] ^C
Interrupt signal received

Traceback (most recent call last):
  File "/usr/lib/python3.4/subprocess.py", line 539, in call
    return p.wait(timeout=timeout)
  File "/usr/lib/python3.4/subprocess.py", line 1566, in wait
    (pid, sts) = self._try_wait(0)
  File "/usr/lib/python3.4/subprocess.py", line 1514, in _try_wait
    (pid, sts) = _eintr_retry_call(os.waitpid, self.pid, wait_flags)
  File "/usr/lib/python3.4/subprocess.py", line 491, in _eintr_retry_call
    return func(*args)
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/powerpill", line 815, in <module>
    sys.exit(run_main())
  File "/usr/bin/powerpill", line 806, in run_main
    return main(args)
  File "/usr/bin/powerpill", line 772, in main
    return powerpill.run_pacman(list(unparse_args(pargs)))
  File "/usr/bin/powerpill", line 701, in run_pacman
    return subprocess.call([self.conf.get('pacman/path')] + args)
  File "/usr/lib/python3.4/subprocess.py", line 541, in call
    p.kill()
  File "/usr/lib/python3.4/subprocess.py", line 1691, in kill
    self.send_signal(signal.SIGKILL)
  File "/usr/lib/python3.4/subprocess.py", line 1681, in send_signal
    os.kill(self.pid, sig)
ProcessLookupError: [Errno 3] No such process
aura >>= Please check your input.
```

**Comment by fosskers** [comment]
Interrupt handling is something that Aura 1 doesn't do at all, so behaviour if done is undefined. Aura 2 is going to be built with this is mind from the beginning.

**Comment by despairblue** [comment]
awesome :+1: , thanks for this amazing tool :)

**Comment by fosskers** [comment]
Np, it's only going to get better.

**Comment by fosskers** [comment]
This should no longer be a problem on `master`, which will be released soon as Aura 2.0.
